src/System.CommandLine.Benchmarks/CommandLine/Perf_Parser_ParseResult.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Benchmarks.Helpers;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+
+namespace System.CommandLine.Benchmarks.CommandLine
+{
+    /// <summary>
+    /// Measures the performance directives parsing.
+    /// </summary>
+    [BenchmarkCategory(Categories.CommandLine)]
+    public class Perf_Parser_ParseResult
+    {
+        private readonly Parser _testParser;
+
+        public Perf_Parser_ParseResult()
+        {
+            var option = new Option("-opt");
+
+            _testParser =
+                new CommandLineBuilder()
+                    .AddOption(option)
+                    .UseParseDirective()
+                    .Build();
+        }
+
+        public IEnumerable<object> GenerateTestParseResults()
+            => new[]
+            {
+                "[directive1] -opt",
+                "[directive1] [directive2] -opt",
+                "[directive1:1] [directive2:2] -opt",
+                "[directive1] [directive2] [directive2] -opt",
+                "[directive1:1] [directive2:2] [directive2:3] -opt",
+            }
+            .Select(cmd => new BdnParam<ParseResult>(_testParser.Parse(cmd), cmd));
+
+        [Benchmark]
+        [ArgumentsSource(nameof(GenerateTestParseResults))]
+        public IDirectiveCollection ParseResult_Directives(BdnParam<ParseResult> parseResult)
+            => parseResult.Value.Directives;
```

Comment:
Sorry, I think I was unclear about the guidance around this and `Diagram` below. As far as I can tell, the [code](https://github.com/dotnet/command-line-api/blob/c3e92bc33b5971d0070627b07e977210c07e597c/src/System.CommandLine/Parser.cs#L70-L82) that's interesting when you use `UseParseDirective` actually happens at the invocation of `Parse`. `Directives` just returns a pre-calculated collection on `ParseResult`, so testing this property getter is uninteresting. If testing calculation and parsing of directives is interesting, the previous test was great for that (removing `.Directives`.)

Reply:
Sorry, I've totally overlooked it.  `=> parseResult.Value.Directives;` is indeed a nonsense. I returned `_testParser.Parse(input).Directives;` back.
Fixed it again.